src/interface/widgets/widget.cpp

```diff
@@ -21,30 +21,32 @@ void Visualizer::paintEvent(QPaintEvent *event) {
     QPainter painter(this);
     painter.setRenderHint(QPainter::Antialiasing, true);
     painter.setRenderHint(QPainter::HighQualityAntialiasing, true);
-    std::optional<rtt::world::view::WorldDataView> world;
-    std::optional<rtt::world::Field> field;
-    auto const &[_, worldPtr] = rtt::world::World::instance();
-    world = worldPtr->getWorld();
-    field = worldPtr->getField();
-
-    if (!world.has_value() || !world.value()->weHaveRobots()) {
-        painter.drawText(24, 24, "Waiting for incoming world state");
-        return;
-    }
 
-    if (field.has_value()) {
-        calculateFieldSizeFactor(field.value());
-        drawBackground(painter);
-        drawFieldHints(field.value(), painter);
-        drawFieldLines(field.value(), painter);
-    }
+    {
+        // worldPtr is scoped in order to release the lock soon as possible
+        auto const &[_, worldPtr] = rtt::world::World::instance();
+        auto const &world = worldPtr->getWorld();
+        auto const &field = worldPtr->getField();
 
-    auto s = QString::fromStdString("We have " + std::to_string(world->getUs().size()) + " robots");
-    painter.drawText(24, 48, s.fromStdString("We have " + std::to_string(world->getUs().size()) + " robots"));
+        if (!world.has_value() || !world.value()->weHaveRobots()) {
```

Comment:
I've always been bothered by the fact that we say we done have a world if we don't have any robots. Could we change this to be shown only when there is no world? If we have no robots, we can still draw the field (and enemy robots)

Reply:
ok I will have look into that